### ISSUE TITLE
Feature: provide options `not create foreign key in mysql`

### ISSUE THREAD

**Issue by day-dreams** [facebook/ent#748]
In some cases, i dont want to use foreigh key,  which will carry some performance cost in MySQL.   Can i use  any option or configuration, to disable foreign keys?

At now, `go generate ./ent` create schemas with foreigh key default.

```golang
// UsersTable holds the schema information for the "users" table.
UsersTable = &schema.Table{
	Name:       "users",
	Columns:    UsersColumns,
	PrimaryKey: []*schema.Column{UsersColumns[0]},
	ForeignKeys: []*schema.ForeignKey{
		{
			Symbol:  "users_groups_users",
			Columns: []*schema.Column{UsersColumns[3]},

			RefColumns: []*schema.Column{GroupsColumns[0]},
			OnDelete:   schema.SetNull,
		},
	},
}
```

**Comment by a8m** [comment]
Hey @day-dreams and thanks for proposing this option. We actually wanted to add this option, but it was blocked on another task that I wanted to add with it.

Basically, ent uses FKs when defining relationships to enforce correctness and consistency on the database side.

For example, if an id of an entity in table A was provided to table B in relationship R, the database checks that the entity actually exists in table A (if not, an error is returned), or if the entity (from table A) that is associated with table B was deleted, the relationship column will be set to NULL automatically by the database (or will be deleted if it's a M2M relationship).

If we add this option, we need to document these caveats in the docs and provide a set of [hooks](https://entgo.io/docs/hooks/#hooks) for implementing this logic. The second part can be added in future PRs.

**Comment by a8m** [comment]
Closed with #751 

Thanks for your contribution @day-dreams 😃
